src/wallet/KintoWalletFactory.sol

```diff
@@ -134,9 +134,18 @@ contract KintoWalletFactory is Initializable, UUPSUpgradeable, OwnableUpgradeabl
     function completeWalletRecovery(address payable wallet, address[] calldata newSigners) external override {
         require(walletTs[wallet] > 0, "invalid wallet");
         require(msg.sender == IKintoWallet(wallet).recoverer(), "only recoverer");
+        require(!kintoID.isKYC(newSigners[0]), "KW-fr: New signer must not be KYC already");
```

Comment:
Suggestion: I think we could probably move this new block (lines from 137 to 139) to the `completeRecovery()` on the `KintoWallet`, I think it's a logic that belongs a bit more to the wallet rather than to the factory, WDYT?

Reply:
no, we cant. At that moment, the id has already been transferred

Reply:
how? if the `transferOnRecovery()` happens on the next line